### PR TITLE
[backup] various fixes

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

fixing issues I found while playing with it on s3.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y
## Test Plan

drilling various scenarios on my local swarm.

### CODE REVIEW

File: execution/executor/src/lib.rs [hunk 1]
```diff
@@ -594,6 +594,10 @@ impl<V: VMExecutor> TransactionReplayer for Executor<V> {
             .update_synced_trees(output.executed_trees().clone());
         Ok(())
     }
+
+    fn expecting_version(&self) -> Version {
+        self.cache.synced_trees().version().map_or(0, |v| v + 1)
```
Comment (lightmark) [inline execution/executor/src/lib.rs:599]: why +1?

Comment (msmouse) [reply execution/executor/src/lib.rs:599]: when you are on a empty db, your are expecting version 0, when you synced to 0, you are expecting 1, right?

**Pull request closed without merging**